### PR TITLE
fix capturing group for matchAll

### LINKED ISSUE #10134: Broken string.matchAll behavior after compilation
## Bug Report

**Current Behavior**
Output code after transpilation doesn't work as expected. It returns undefined capture groups. See below.

**Input Code**
I took last example from here: https://v8.dev/features/string-matchall ([repl](https://babeljs.io/en/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYewdgzgLgBNBOBLMBzGBeGByAYgQwDcQkoBTGAcUSgAkBXAIxnlIAcQIAuGKYAZgCcAelLAAtngBMANkkwCADiGKAdABNSBLAG4AUKEiwWKUgA8MMIQB0GACgD8AHhAB3MKXgA-ANp4AtABeAAx-AgC6ANQAlFZCDo4s7D7-waFWKpExDEIoegBmxDC2BtAwErwAFjAgeXBQSKgq5cAVAIIANu22xmZRUTAA3rowMCUg7aQq7SAotgAGACQDzRXeQWEAvjB4sEsrKsgaplsu1FVYezstB2CsdFAbWHNReiNjE1Mz84BJhNVuHtxLpUVCh4CA6KwICpXO54Btnq9RuAIONJtNZnNfokQIDllcKiCwRCodj4S9dBsgA&debug=true&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env&prettier=false&targets=Node-9&version=7.4.5&externalPlugins=))

```js
const string = 'Favorite GitHub repos: tc39/ecma262 v8/v8.dev';
const regex = /\b(?<owner>[a-z0-9]+)\/(?<repo>[a-z0-9\.]+)\b/g;
for (const match of string.matchAll(regex)) {
  console.log(`${match[0]} at ${match.index} with '${match.input}'`);
  console.log(`→ owner: ${match.groups.owner}`);
  console.log(`→ repo: ${match.groups.repo}`);
}
```

It returns this when you run it just like this before Babel:

```text
tc39/ecma262 at 23 with 'Favorite GitHub repos: tc39/ecma262 v8/v8.dev'
→ owner: tc39
→ repo: ecma262
v8/v8.dev at 36 with 'Favorite GitHub repos: tc39/ecma262 v8/v8.dev'
→ owner: v8
→ repo: v8.dev
```

**Expected behavior/code**
It should return the same output. Unfortunately, it returns this:

```text
tc39/ecma262 at 23 with 'Favorite GitHub repos: tc39/ecma262 v8/v8.dev'
→ owner: undefined
→ repo: undefined
v8/v8.dev at 36 with 'Favorite GitHub repos: tc39/ecma262 v8/v8.dev'
→ owner: undefined
→ repo: undefined
```

(capturing groups got lost and return undefined values after translation)

**Babel Configuration (.babelrc, package.json, cli command)**

Reproducible in [REPL](https://babeljs.io/en/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYewdgzgLgBNBOBLMBzGBeGByAYgQwDcQkoBTGAcUSgAkBXAIxnlIAcQIAuGKYAZgCcAelLAAtngBMANkkwCADiGKAdABNSBLAG4AUKEiwWKUgA8MMIQB0GACgD8AHhAB3MKXgA-ANp4AtABeAAx-AgC6ANQAlFZCDo4s7D7-waFWKpExDEIoegBmxDC2BtAwErwAFjAgeXBQSKgq5cAVAIIANu22xmZRUTAA3rowMCUg7aQq7SAotgAGACQDzRXeQWEAvjB4sEsrKsgaplsu1FVYezstB2CsdFAbWHNReiNjE1Mz84BJhNVuHtxLpUVCh4CA6KwICpXO54Btnq9RuAIONJtNZnNfokQIDllcKiCwRCodj4S9dBsgA&debug=true&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env&prettier=false&targets=Node-9&version=7.4.5&externalPlugins=) with env preset and Node.js <9 (latest Node.js are broken in REPL).

### PR DESCRIPTION
**Pull request by tanhauhau**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #10134
| Patch: Bug Fix?          | 👍 
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

According to spec (https://github.com/tc39/proposal-string-matchall/blob/master/spec.md#stringprototypematchall--regexp-), when calling `String.matchAll`, the regex is recreated again with 'g' flag, and the current `BabelRegExp` constructor will take the 'g' flag as group descriptor.

I see there's no `String.prototype.matchAll` support in node.js, not sure how to test this

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
In node 12 it is supported: you can use `minNodeVersion` in the test options.
Also, we should have another fixture to test it loading the `core-js/features/string/match-all` polyfill.

**Comment by tanhauhau** [comment]
@nicolo-ribaudo apparently we dont have node 12 for the test env, https://github.com/babel/babel/blob/master/.travis.yml#L8 . somehow, it's supposed to be added here https://github.com/babel/babel/commit/80a5a2e7ddcaa034991cc46b6342b2be3266f46a, but the node env for travis is node 11, not node 12

**Comment by jridgewell** [comment]
We can add node 12 to the Travis yaml

**Comment by existentialism** [comment]
@tanhauhau we test node 12 via _circle_:

https://github.com/babel/babel/blob/master/.circleci/config.yml#L37

**Comment by tanhauhau** [comment]
@nicolo-ribaudo how do i add `core-js/features/string/match-all` polyfill into the fixture?

**Comment by nicolo-ribaudo** [comment]
Doesn't adding it to the dev dependencies and then `require`ing it work?

**Comment by mrtnzlml** [comment]
Hi, thanks so much for taking care of my reported issue. Is there anything I can do to help with moving it forward?

**Comment by tanhauhau** [comment]
@mrtnzlml thanks for nudging, @nicolo-ribaudo i've added another fixture that test with core-js@v3 String.matchAll polyfill

File: packages/babel-plugin-transform-named-capturing-groups-regex/package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   "repository": "https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-named-capturing-groups-regex",
   "bugs": "https://github.com/babel/babel/issues",
   "dependencies": {
+    "core-js-pure": "^3.0.0",
```
Comment (existentialism) [inline packages/babel-plugin-transform-named-capturing-groups-regex/package.json]: Shouldn't this be a `devDependency`?

Comment (tanhauhau) [reply packages/babel-plugin-transform-named-capturing-groups-regex/package.json]: oops.. updated @existentialism

**Comment by nicolo-ribaudo** [comment]
I tried to use GitHub's "delete file" feature, thinking that it would have reverted the changes to a single file. I messed it up :sweat_smile:

**Comment by mrtnzlml** [comment]
Hi @nicolo-ribaudo, me again... 😊 Would you please consider merging this PR into master?

**Comment by nicolo-ribaudo** [comment]
@mrtnzlml We have a two-:heavy_check_mark:s policy for merging PRs; I'll try to find some reviewers!

**Comment by existentialism** [comment]
I was just waiting on the revert of the package.json to +1

**Comment by tanhauhau** [comment]
Sorry my bad. fixed the wrong changes and reverts on `package.json`

**Pull request merged**